ghostratings/calculations.php

```diff
@@ -0,0 +1,346 @@
+                   {
+                     $actualResult[$userID] = 0;
+                   }
+                 }
+                 $grAdjustment[$userID] = (($grSum / $this->modvalue) * ($actualResult[$userID]-$expectedResult[$userID]));
+               }
+               break;
+             //In this case we assume that all 1v1 games are unranked, and the previous code excludes unranked non-1v1 games from being calculated, so this is the 1v1 calculation 
+             case "Unranked":
+               $first = True;
+               $r1 = 0;
+               $r2 = 0;
+               $id1 = 0;
+               $id2 = 0;
+               $result = 0;
+               foreach ($userGR as $userID => $rating)
+               {
+                 if($first)
+                 {
+                   $r1 = $rating;
+                   $id1 = $userID;
+                   $first = False;
+                 }
+                 else
+                 {
+                   $r2 = $rating;
+                   $id2 = $userID;
+                 }
+               }
+               if ($this->gameStatus == "Drawn")
+               {
+                 $result = 0.5;
+               }
+               else if ($this->winner == $id1)
+               {
+                 $result = 1;
+               }
+               $R1 = pow(10,($r1/400));
+               $R2 = pow(10,($r2/400));
+               $E1 = $R1 / ($R1 + $R2);
+               $E2 = $R2 / ($R1 + $R2);
+               $grAdjustment[$id1] = $this->k * ($result - $E1);
+               $grAdjustment[$id2] = $this->k * (1 - $result - $E2);
+               break;
+           }
+           foreach ($grAdjustment as $userID=>$adjustment)
+           {
+             $newRating = $userGR[$userID] + $adjustment;
+             $sqlUpdate = "UPDATE wD_GhostRatings SET rating=".$newRating;
+             if ($newRating > $peakGR[$userID])
+             {
+               $sqlUpdate = $sqlUpdate . ", peakRating=".$newRating;
+             }
+             if ($date <> $monthYear[$userID])
+             {
+               $sqlUpdate = $sqlUpdate. ", monthYear=".$date;
+               $DB->sql_put("INSERT INTO wD_GhostRatingsHistory(userID, categoryID, monthYear, rating) VALUES(".$userID.", ".$categoryID.", ".$date.", ".$newRating.")");
+             }
+             else
+             {
+               $DB->sql_put("UPDATE wD_GhostRatingsHistory SET rating=".$newRating." WHERE userID=".$userID." AND categoryID=".$categoryID." AND monthYear=".$date);
+             }
+             $sqlUpdate = $sqlUpdate . " WHERE categoryID=".$categoryID." AND userID=".$userID;
+             $DB->sql_put($sqlUpdate);
+             $DB->sql_put("INSERT INTO wD_GhostRatingsBackup(userID, categoryID, gameID, adjustment, timeFinished) VALUES(".$userID.", ".$categoryID.", ".$this->gameID.", ".$adjustment.", ".$this->time.")");
+           }
+         }
+       }
+     }
+     $DB->sql_put("UPDATE wD_Games SET grCalculated=1 WHERE id=".$this->gameID);
```

Comment:
add limit 1 for safety.

Reply:
Just put in a commit which addresses this.